Add information for site-to-site connectivity (#10508)
circleci/circleci-docs: docs/guides/modules/security/pages/site-to-site-connectivity.adoc

= Site-to-site connectivity

> :page-platform: Cloud

Reviewer: ```suggestion
:page-badge: Preview
:page-platform: Cloud
```

:page-description: Run CircleCI Cloud jobs that reach private infrastructure without opening your firewall to the internet.
:experimental:

> NOTE: Site-to-site connectivity is available on link:https://circleci.com/pricing/[CircleCI Scale plans] only. Contact your account team to confirm eligibility before getting started.

Reviewer: ```suggestion
NOTE: Site-to-site connectivity is available on link:https://circleci.com/pricing/[CircleCI Scale plan] only. Contact your account team to confirm eligibility before getting started.
```

[#overview]

> == Overview

Reviewer: ```suggestion
```

> Site-to-site connectivity lets CircleCI Cloud jobs securely reach resources behind your firewall, like a private GitHub Enterprise instance, an internal Artifactory, or on-prem APIs. This removes the need to expose those resources to the public internet or add CircleCI's IP ranges to your network allow list.

Reviewer: ```suggestion
Site-to-site connectivity lets CircleCI Cloud jobs securely reach resources behind your firewall, like a private GitHub Enterprise instance, an internal Artifactory, or on-prem APIs. Site-to-site connectivity removes the need to expose private resources to the public internet or add CircleCI's IP ranges to your network allow list.
```

[#how-it-works]
== How it works

Site-to-site connectivity creates an encrypted tunnel between a lightweight endpoint you run in your network and your CircleCI jobs. When the tunnel connects, jobs reach your internal resources directly.

> You start the connection via the link:https://github.com/CircleCI-Labs/site-to-site-connectivity-orb[site-to-site connectivity orb]. CircleCI manages the tunnel service on its end.

Reviewer: Does it make sense to link to the orb source rather than the orb page in our orb registry?
```suggestion
You start the connection by using the link:https://github.com/CircleCI-Labs/site-to-site-connectivity-orb[site-to-site connectivity orb] in your CircleCI configuration. CircleCI manages the tunnel service on its end.
```

[#what-you-can-do]
== What you can do today

Once a job is running with the orb configured, it can:

* *Clone from private GHES or GitLab* - SSH-based clones from a self-hosted GitHub Enterprise or GitLab instance work out of the box.
* *Call internal HTTPS endpoints* - Reach internal package registries, REST APIs, Artifactory, or any service on your internal network.
* *Pull images during a job* - Run `docker pull` as a job step against a private registry over the tunnel.
* *Preserve end-to-end TLS* - The job respects your destination's TLS certificate with no man-in-the-middle.
* *Route wildcard host names* - Route an entire domain (for example, `+*.internal.acme.dev+`) through a single tunnel setup.

[#considerations]
== Considerations

*Guided onboarding required* - CircleCI runs setup with you end-to-end: scoping your use case, configuring the tunnel, and verifying the connection.

> *Docker executor support* - Because the executor fetches Docker images before the tunnel starts, your registry must support OIDC. See xref:permissions-authentication:openid-connect-tokens.adoc[OpenID Connect Tokens] for more information.

Reviewer: "the executor" - does this mean the executor defined in the site to site connectivity orb?

What is "your registry" in this context? And by support are we saying you need to go to that guide and set up OIDC or check that something is enabled?

Author: I struggled to refine the line and hoping this is clearer.

> *Docker executor support* - CircleCI fetches your executor's Docker image before the tunnel starts. If that image lives in a private registry, configure OIDC between CircleCI and your registry so CircleCI can pull it. 

"Executor" refers to the Docker executor. "Your registry" is the customer's registry of Docker images, and I'm trying to clarify that they need to set up OIDC first so we can fetch the image from their image registry.

[#prerequisites]
== Prerequisites

. A CircleCI Scale plan.

> . An agent deployment inside your network, managed by your team.

Reviewer: Not sure what an "agent deployment" means here

Author: I rephrased it to 

> A tunnel endpoint inside your network, deployed and managed by your team.

. Time to work with CircleCI during onboarding.

> . A use case that maps to the capabilities described in this page.

Reviewer: ```suggestion
. A use case that maps to the capabilities described on this page.
```

[#is-this-right-for-you]
== Is this right for you?

[.table.table-striped]
[cols="1,1", options="header"]
|===
| Good fit
| Not a fit today

> | Running GHES or GitLab on-prem and unwilling to publish it externally

Reviewer: "unwilling" maybe a bit too negative, "unable"?

Author: Changed to "unable".
